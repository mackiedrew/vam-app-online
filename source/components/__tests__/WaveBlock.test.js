import Subject from "../WaveBlock";

const mockProps = {
  amplitude: 3,
  maxAmplitude: 10
};

describe("<WaveBlock /> structure", () => {
  const wrapper = shallow(<Subject {...mockProps} />);

  it("renders without crashing", () => {
    shallow(<Subject {...mockProps} />);
  });

  it("renders as a div with class: `wave-block`", () => {
    expect(wrapper.is("div.wave-block")).toEqual(true);
  });

  it("renders a single button with class: `amplitude`", () => {
    expect(wrapper.find("button.amplitude")).toHaveLength(1);
  });

  it("renders without crashing when quiet", () => {
    shallow(<Subject {...mockProps} quiet />);
  });

  it("renders without crashing when selected", () => {
    shallow(<Subject {...mockProps} selected />);
  });

  it("renders without crashing when filler", () => {
    shallow(<Subject {...mockProps} filler />);
  });

  it("renders without crashing when more", () => {
    shallow(<Subject {...mockProps} more />);
  });
});

describe("<WaveBlock /> display math", () => {
  it("relative amplitude is returns 1 (100%) when provided with maxAmplitude === amplitude", () => {
    const wrapper = shallow(<Subject amplitude={10} maxAmplitude={10} />);
    expect(wrapper.instance().relativeAmplitude()).toEqual(1);
  });

  it("relative amplitude is returns 0 (0%) when provided with amplitude 0", () => {
    const wrapper = shallow(<Subject amplitude={0} maxAmplitude={10} />);
    expect(wrapper.instance().relativeAmplitude()).toEqual(0);
  });

  it("amplitudeStyle returns an object with expected keys", () => {
    const wrapper = shallow(<Subject amplitude={5} maxAmplitude={10} />);
    expect(wrapper.instance().amplitudeStyle()).toBeInstanceOf(Object);
    expect(wrapper.instance().amplitudeStyle().height).toEqual("50%");
  });
});

describe("<WaveBlock /> handleClick()", () => {
  it("calls seekTo() function one time", () => {
    const mockSeekTo = sinon.spy();
    const wrapper = shallow(<Subject {...mockProps} seekTo={mockSeekTo} />);
    wrapper.instance().handleClick();
    expect(mockSeekTo.called).toEqual(true);
  });
});
