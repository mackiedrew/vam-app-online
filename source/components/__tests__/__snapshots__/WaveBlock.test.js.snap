// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WaveBlock /> renders correctly with bare minimum props 1`] = `
<div
  className="wave-block"
  style={
    Object {
      "backgroundColor": "rgb(240, 240, 240)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "0.4",
      }
    }
  />
</div>
`;

exports[`<WaveBlock /> renders correctly with bare minimum props, and filler grain 1`] = `
<div
  className="wave-block filler"
  style={
    Object {
      "backgroundColor": "rgb(240, 240, 240)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    track ends
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "0.4",
      }
    }
  />
</div>
`;

exports[`<WaveBlock /> renders correctly with bare minimum props, and grain with amplitude 1`] = `
<div
  className="wave-block"
  style={
    Object {
      "backgroundColor": "rgb(240, 240, 240)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "0.4",
      }
    }
  />
</div>
`;

exports[`<WaveBlock /> renders correctly with bare minimum props, and maxAmplitude 1`] = `
<div
  className="wave-block"
  style={
    Object {
      "backgroundColor": "rgb(240, 240, 240)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "0.4",
      }
    }
  />
</div>
`;

exports[`<WaveBlock /> renders correctly with bare minimum props, and more grain 1`] = `
<div
  className="wave-block more"
  style={
    Object {
      "backgroundColor": "rgb(240, 240, 240)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    more...
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "0.4",
      }
    }
  />
</div>
`;

exports[`<WaveBlock /> renders correctly with bare minimum props, and quiet grain 1`] = `
<div
  className="wave-block"
  style={
    Object {
      "backgroundColor": "rgb(240, 230, 230)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "0.4",
      }
    }
  />
</div>
`;

exports[`<WaveBlock /> renders correctly with bare minimum props, and selected 1`] = `
<div
  className="wave-block"
  style={
    Object {
      "backgroundColor": "rgb(240, 240, 240)",
      "flexGrow": 13,
    }
  }
>
  <div
    className="note"
  >
    
  </div>
  <button
    className="amplitude"
    onClick={[Function]}
    style={
      Object {
        "height": "0%",
        "opacity": "1.0",
      }
    }
  />
</div>
`;
